### DISCUSSION TITLE
Ordering of images in masonry / rows layout

### DISCUSSION THREAD

**Discussion by joshkay** [igordanchenko/react-photo-album#171] [Q&A] [answered]
I currently have a requirement to have the images sorted by a date field, but it looks like neither the rows layout or the masonry layout arrange the images according to the order of the array provided. 

This behavior is completely understandable as rearranging the images to fill space seems to be one of the main purposes of this library.  

Is it possible to have either of these layouts sorted or should I just create my own layout with css grid / flexbox?

**Comment by igordanchenko** [comment, marked as answer]
That’s not actually the case when you use `rows` layout. In rows layout the photos are arranged in the exact order they are provided in the `photos` array. Can you take another look at the rows layout?

**Reply by joshkay** [reply]
My mistake thanks for the quick response!
